File: 0566-reshape-the-matrix/0566-reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        matrix = [[0 for col in range(c)] for row in range(r)]
        arr = []
        m = len(mat)
        n = len(mat[0])
        if m*n == r*c:
            for row in range(m):
                for col in range(n):
                    arr.append(mat[row][col])

            index = 0
            for i in range(r):
                for j in range(c):
                    matrix[i][j] = arr[index]
                    index+=1

                    
        else:
            return mat
        
        return matrix

       
        

File: 0566-reshape-the-matrix/0566-reshape-the-matrix.java
class Solution {
    public int[][] matrixReshape(int[][] mat, int r, int c) {
        int m = mat.length;
        int n =mat[0].length;
        if (m*n != r*c){
            return mat;
        }
        ArrayList<Integer> arr = new ArrayList<>();
        for(int row=0; row<m; row++){
            for(int col=0; col<n; col++){
                arr.add(mat[row][col]);
            }
        }
        int[][] matrix = new int[r][c];
        int index = 0;
        for (int i =0; i<r;i++){
            for (int j =0;j<c; j++){
                matrix[i][j] = arr.get(index);
                index++;
            }
        }
        return matrix;
        
    }
}